tests/test_e2e.py

```diff
@@ -223,6 +223,29 @@ def _test_username_password(self,
             )
         return result
 
+    def _test_iwa(self,
+                  authority=None, client_id=None, username=None, scope=None,
+                  client_secret=None,
+                  azure_region=None,
+                  http_client=None,
+                  **ignored):
+        assert authority and client_id and username and scope
+        self.app = self._build_app(
+            client_id, authority=authority,
+            http_client=requests,
+            azure_region=azure_region, client_credential=client_secret)
+        self.assertEqual(
+            self.app.get_accounts(username=username), [], "Cache starts empty")
+        result = self.app.acquire_token_integrated_windows_auth(
```

Comment:
@iulico-1 - how is this going to work on Linux, where there is no WAM? We can't expect every customer to enroll their device either.

Reply:
On linux there is no WIA also :) this api will just fail with UIRequired on linux without broker integration.

Reply:
> On linux there is no WIA also :) this api will just fail with UIRequired on linux without broker integration.

But the customer [claimed otherwise](https://github.com/AzureAD/microsoft-authentication-library-for-python/pull/652#issuecomment-2427190084).

Reply:
A form of WIA can be setup on Linux.

Reply:
So, am I understanding correctly that we now agree on there is a need for Integrated Windows Auth (IWA, a.k.a. WIA) for Linux? And then this conversation is back to the original question, i.e. how the IWA shall be exposed in an API? Brainstorming time. :-)

1. This PR's current implementation `acquire_token_integrated_windows_auth(username, scopes)` was probably inspired by [MSAL .Net's dedicated IWA api](https://learn.microsoft.com/en-us/entra/msal/dotnet/acquiring-tokens/desktop-mobile/integrated-windows-authentication).
  If we do not want IWA to be a dedicated API, we will need to fit it into other existing APIs. Such as:
2. Modeling it as a ROPC but without a password `acquire_token_by_username_password(username, password=None, scopes=[...])`. This makes sense from the implementation standpoint, because the actual implementation already shares many building blocks (mex.py, wstrust_request.py) which are only used in ROPC code path.
3. Modeling it as "the same pattern we did with OS account on ~.net~ Python", and that is `acquire_token_interactive(scopes, login_hint="johndoe@example.com", prompt="none")`

Which one do we prefer?

Reply:
@rayluo @iulico-1 I think this fits best in the `acquire_token_interactive` flow, and the library can then determine that it's IWA based on parameter heuristics. This will also help reduce the API surfaces that customers have to navigate to use the capability, _and_ simplify broker intergration.